android-kit-base/src/main/java/com/mparticle/kits/KitManagerImpl.java

```diff
@@ -1361,13 +1375,61 @@ public void execute(String viewName,
                             placeHolders,
                             fontTypefaces,
                             FilteredMParticleUser.getInstance(user.getId(), provider));
+                    });
                 }
             } catch (Exception e) {
                 Logger.warning("Failed to call execute for kit: " + provider.getName() + ": " + e.getMessage());
             }
         }
     }
 
+    private void confirmEmail(
+            @Nullable String email,
+            @Nullable MParticleUser user,
+            MParticle instance,
+            Runnable runnable
+    ) {
+        if (email != null && user != null) {
+            String existingEmail = user.getUserIdentities().get(MParticle.IdentityType.Email);
+
+            if (!email.equals(existingEmail)) {
+                // If there's an existing email but it doesn't match the passed-in email, log a warning
+                if (existingEmail != null) {
+                    Logger.warning( String.format(
+                            "The existing email on the user (%s) does not match the email passed to selectPlacements (%s). " +
+                                    "Please make sure to sync the email identity to mParticle as soon as it's available. " +
+                                    "Identifying user with the provided email before continuing to selectPlacements.",
+                            existingEmail, email
+                    ));
+                }
+
+                IdentityApiRequest identityRequest = IdentityApiRequest.withUser(user)
+                        .email(email)
+                        .build();
+                MParticleTask<IdentityApiResult> task = instance.Identity().identify(identityRequest);
```

Comment:
I see you're passing in the entire `instance` to just use Identity. Would it be cleaner to just pass in the `identity` service by itself?

I know we do this a lot in our SDKs, where we pass in the entire mPInstance object, but we talked about this being an antipattern that we wanted to avoid.